### PR TITLE
Added my spaces page per UXD design

### PR DESCRIPTION
**Pull request by dlabrecq**
Added my spaces page per the UXD design below:
https://redhat.invisionapp.com/share/DQDSL22JA#/screens/261389228

The my spaces page uses the new pinning feature from the patternfly-ng list. The intention is that pinned items persist across sessions. They are also the primary sort, meaning they are always shown at the top of the list.

Note that in order for the toolbar to sort and filter properly, the list component does not use infinite scroll. Unfortunately, the existing namespaces API does not support a server-side sort and filter.

Works best if you have a few spaces to play with.

Caveats:
1. No API to display spaces which the user is a collaborator on
2. No API to show activity
3. No API to show private spaces

Will file issues to track the API requirements separately.

screen shot 2017-12-11 at 2 51 27 pm

### CODE REVIEW

**Comment by fabric8cd** [comment]
@dlabrecq fabric8/fabric8-ui:SNAPSHOT-PR-2304-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2304-fabric8-ui.badger.fabric8.io

**Comment by dlabrecq** [comment]
Below is my wish list of issues needed to implement this UXD design properly.

- Need API to show where users is a collaborator: https://github.com/openshiftio/openshift.io/issues/1630
- Need API to show activity: https://github.com/openshiftio/openshift.io/issues/1631
- Need API to show private spaces: https://github.com/openshiftio/openshift.io/issues/1632
- API: /namedspaces/{userName} to support sort and filter: https://github.com/openshiftio/openshift.io/issues/1627
- API: /namedspaces/{userName} to support all spaces: https://github.com/openshiftio/openshift.io/issues/1628
- API: /search/spaces to support sort and filter: https://github.com/openshiftio/openshift.io/issues/1629

FYI, @catrobson and @bdellasc

File: src/app/layout/header/header.component.html [hunk 1]
```diff
@@ -88,6 +88,14 @@
                   Create space
                 </a>
               </li>
+              <li id="header_mySpaces">
+                <a [routerLink]="[loggedInUser.attributes.username,'_myspaces']">
+                <span class="nav-item-icon">
```
Comment (mindreeper2420) [inline src/app/layout/header/header.component.html]: indent the `<span>`

File: src/app/profile/my-spaces/my-spaces-item/my-spaces-item.component.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<div class="list-pf-content-wrapper" *ngIf="space?.relationalData?.creator?.attributes">
+  <div class="list-pf-main-content">
+    <div class="list-pf-title">
+      <span>
```
Comment (mindreeper2420) [inline src/app/profile/my-spaces/my-spaces-item/my-spaces-item.component.html]: I don't believe this `<span>` is needed as a wrapper to the `<a>`

File: src/app/profile/my-spaces/my-spaces-item/my-spaces-item.component.less [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@import (reference) '../../../../assets/stylesheets/shared/osio.less';
+.my-spaces-page {
+  .avatar {
+    border-left: 1px solid @color-pf-black-300;
```
Comment (mindreeper2420) [inline src/app/profile/my-spaces/my-spaces-item/my-spaces-item.component.less]: for OSIO, we break these out:

File: src/app/profile/my-spaces/my-spaces.component.html [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+<ng-template #deleteSpace title="Remove the space">
+  <div class="modal-header">
+    <button class="close" aria-label="Close" type="button"
+            (click)="closeModal()">
+      <span aria-hidden="true">&times;</span>
+    </button>
+    <dt class="modal-title">Remove Space</dt>
+  </div>
+  <div class="modal-body">
+    <h2>Are you sure you want to remove the space</h2>
```
Comment (mindreeper2420) [inline src/app/profile/my-spaces/my-spaces.component.html]: missing a `?` at the end

File: src/app/profile/my-spaces/my-spaces.component.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<div class="my-spaces-page">
+  <my-spaces-toolbar
+      (onCreateSpace)="openForgeWizard(createSpace)"
+      (onFilterChange)="filterChange($event)"
+      (onSortChange)="sortChange($event)"
+      [resultsCount]="resultsCount">
+  </my-spaces-toolbar>
+  <div class="container-fluid margin-bottom-20">
+    <div class="row">
+      <div class="col-md-12">
```
Comment (mindreeper2420) [inline src/app/profile/my-spaces/my-spaces.component.html]: should this be `col-sm-12`, in order to fit smaller buckets?

**Comment by mindreeper2420** [comment]
@dlabrecq There is a `1px` difference in the width of the empty space to the left of the Space name and when the pin is in place (38px when empty vs 39px with pin). This is causing a jump to occur when the pin is added/removed.

**Comment by fabric8cd** [comment]
@dlabrecq fabric8/fabric8-ui:SNAPSHOT-PR-2304-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2304-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@dlabrecq fabric8/fabric8-ui:SNAPSHOT-PR-2304-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2304-fabric8-ui.badger.fabric8.io

**Pull request merged**